cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,6 +40,7 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+	SelfHosted        bool
```

Comment:
I expect nearly everything here will be completely rewritten in future releases -- in line with my phases proposal for example. Let's discuss the proposed, long-term goals there and then implement that for future releases